src/DateTimePicker.jsx

```diff
@@ -98,15 +113,20 @@ class DateTimePicker extends React.Component {
 		return (datetime.toTimeString().split(' ')[0]).split(':', 2).join(':'); // TODO localize toLocaleTimeString ?
 	}
 
+	parseNewTimeAsDate(timeStr) {
```

Comment:
This should get some docs to explain what the expect input-output formats are.

It looks like you just need to set the 'time' value on a particular 'date' object - you should be able to use the [Date object Setters](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Date) to do that - since it looks like `timeStr` will represent the local timezone, parse it and use `setHours` and `setMinutes` on a `new Date(this.state.datetime)`.

If you really really need to get and use a timezone offset, use [`getTimezoneOffset`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset), but I'm hoping you won't need to.